src/mission.rs

```diff
@@ -6,24 +6,37 @@ use crate::geo::orientation::Orientation;
 use crate::robot::Command;
 use crate::robot::Robot;
 
-#[derive(Debug, Eq, PartialEq)]
-pub struct Mission {
+#[derive(Debug, PartialEq)]
+pub enum Outcome {
+    Success(Robot),
+    Lost(Robot),
+}
+
+pub struct Mission<I, X>
+where
+    I: Iterator<Item = X>,
+{
     pub upper_right: Point,
+    source: I,
     scents: HashMap<Point, HashSet<Orientation>>,
 }
 
 const ORIGIN: Point = Point { x: 0, y: 0 };
 
-impl Mission {
-    pub fn new(upper_right: Point) -> Mission {
+impl<I, SourceItem> Mission<I, SourceItem>
+where
+    I: Iterator<Item = SourceItem>,
+{
+    pub fn new(upper_right: Point, source: I) -> Mission<I, SourceItem> {
         Mission {
             upper_right: upper_right,
```

Comment:
Not super important, but if the source & destination names are the same you can just do:

```suggestion
            upper_right,
```